config.go

```diff
@@ -211,21 +211,20 @@ func getPluginVersion() string {
 		- Machine Identities.
 */
 func isValidIAM(arn *string, client *alks.Client) bool {
-	// Check if Admin || IAMAdmin
-	if strings.Contains(*arn, "assumed-role/Admin/") || strings.Contains(*arn, "assumed-role/IAMAdmin/") || strings.Contains(*arn, "assumed-role/LabAdmin/") {
-		return true
-	}
 
-	// Check if MI...
-	arnParts := strings.FieldsFunc(*arn, splitBy)
-	iamArn := fmt.Sprintf("arn:aws:iam::%s:role/acct-managed/%s", arnParts[3], arnParts[5])
+	// Validate Machine Identity
+	responseMI, err := client.IsIamEnabled(*arn)
+	if err != nil {
+		return false
+	}
```

Comment:
I believe we can drop this check.  All credentials end up being converted to STS before we get to this point, so I believe we can just make a bare call and validate them that way.  Might need to double check how things are implemented in ALKS, but it's my understand that should work.

Reply:
Gotcha I see it now. I'm going to actually clean it up further in that case and remove the method and make this a 2 line call to simplify. 👍🏼